.github/workflows/macos_tests.yml

```diff
@@ -367,7 +367,7 @@ jobs:
         run: /usr/bin/xcodebuild -quiet -scheme ${BUILD_SCHEME} -destination "generic/platform=visionos" build
       - name: visionOS test  # arm only
         if: '!cancelled() && inputs.visionos_xcode_test_enabled'
-        run: /usr/bin/xcodebuild -quiet -scheme ${BUILD_SCHEME} -destination "name=Apple Vision Pro" test
+        run: TARGET_OS=""; if [ "$XCODE_VERSION" = "16.2" ]; then TARGET_OS=",OS=2.3"; fi && /usr/bin/xcodebuild -quiet -scheme ${BUILD_SCHEME} -destination "name=Apple Vision Pro$TARGET_OS" test
```

Comment:
This only rights on the scheduled CI jobs so this was tested by altering the PR job temporarily: https://github.com/apple/swift-nio/pull/3307/commits/cb1fd1a3ccc9b57b5dd49f112361dd16be538fe0

Here's the test run: https://github.com/apple/swift-nio/actions/runs/16341431397